packages/teleport/src/Discover/Database/DownloadScript/DownloadScript.tsx

```diff
@@ -0,0 +1,378 @@
+          fallbackFn={props => (
+            <Box>
+              <Heading />
+              <Labels labels={labels} setLabels={setLabels} />
+              <ButtonSecondary width="200px" onClick={props.retry}>
+                Generate Command
+              </ButtonSecondary>
+              <Box>
+                <TextIcon mt={3}>
+                  <Icons.Warning ml={1} color="danger" />
+                  Encountered Error: {props.error.message}
+                </TextIcon>
+              </Box>
+              <ActionButtons onProceed={() => null} disableProceed={true} />
+            </Box>
+          )}
+        >
+          <Suspense
+            fallback={
+              <Box>
+                <Heading />
+                <Labels
+                  labels={labels}
+                  setLabels={setLabels}
+                  disableAddBtn={true}
+                />
+                <ButtonSecondary width="200px" disabled={true}>
+                  Generate Command
+                </ButtonSecondary>
+                <ActionButtons onProceed={() => null} disableProceed={true} />
+              </Box>
+            }
+          >
+            {!showScript && (
+              <Box>
+                <Heading />
+                <Labels labels={labels} setLabels={setLabels} />
+                <ButtonSecondary
+                  width="200px"
+                  onClick={() => handleGenerateCommand(validator)}
+                >
+                  Generate Command
+                </ButtonSecondary>
+                <ActionButtons onProceed={() => null} disableProceed={true} />
+              </Box>
+            )}
+            {showScript && (
+              <DownloadScript
+                {...props}
+                labels={labels}
+                setLabels={setLabels}
+              />
+            )}
+          </Suspense>
+        </CatchError>
+      )}
+    </Validation>
+  );
+}
+
+export function DownloadScript(
+  props: AgentStepProps & {
+    labels: AgentLabel[];
+    setLabels(l: AgentLabel[]): void;
+  }
+) {
+  // Fetches join token.
+  const { joinToken, reloadJoinToken, timeout } = useJoinToken(
+    ResourceKind.Database,
+    props.labels
+  );
+
+  // Starts resource querying interval.
+  const {
+    timedOut: pollingTimedOut,
+    start,
+    result,
+  } = usePingTeleport<Database>();
+
+  function regenerateScriptAndRepoll() {
+    reloadJoinToken();
+    start();
+  }
+
+  function handleNextStep() {
+    props.updateAgentMeta({
+      ...props.agentMeta,
+      resourceName: result.name,
+      db: result,
+    });
+  }
+
+  let poll: Poll = { state: 'polling' };
+  if (pollingTimedOut) {
+    poll = {
+      state: 'error',
+      error: {
+        reasonContents: [
+          <>
+            The command was not run on the server you were trying to add,
+            regenerate command and try again.
+          </>,
+          // TODO (lisa): not sure what this message should be.
```

Comment:
Can you create an issue to track this update as this will likely get forgotten about otherwise.

Reply:
tracked here https://github.com/gravitational/teleport/issues/16858